tools/CheckDates.scala

```diff
@@ -0,0 +1,184 @@
+    val messages = buf_messages.split("commit").distinct.drop(1)
+    val dates = buf_dates.split("\\n").distinct
+    val gits_info = (messages, dates).zipped.map{ case(a,b) => GitInfo.tupled((a,b))}
+    val res = gits_info.filter(!_.message.contains("[Dates fixing]"))
+    res(0).date.split(" ")(4)
+  }
+
+  def getYear(pathToFile: String, lower_bound: Int, upper_bound: Int): String = {
+    val file = readFile(pathToFile)
+    val start_index = file.indexOf("(c) 20")
+    val year = file.substring(start_index + lower_bound, start_index + upper_bound)
+    year
+  }
+
+  def getActualCreationYear(pathToFile: String): String = {
+    val lb = "(c) ".length()
+    val ub = "(c) 20XX".length()
+    val actualCreationYear = getYear(pathToFile, lb, ub)
+    actualCreationYear
+  }
+
+  def getActualLastUpdateYear(pathToFile: String): String = {
+    val lb = "(c) 20XX-".length()
+    val ub = "(c) 20XX-XXXX".length()
+    val actualLastUpdateYear = getYear(pathToFile, lb, ub)
+    actualLastUpdateYear
+  }
+
+  def analyzeDates(): Array[FileDates] = {
+    val files_array = readFilesList()
+    var data = Array[FileDates]()
+    for(file_path <- files_array) {
+      if(readFile(file_path).contains("|    (c) 20") && ifFileWasInGit(file_path)) {
+        val actualLastUpdateYear = getActualLastUpdateYear(file_path)
+        val gitLastUpdateYear = getGitLastUpdateYear(file_path)
+        val actualCreationYear = getActualCreationYear(file_path)
+        val gitCreationYear = getGitCreationYear(file_path)
+        data = data :+ FileDates.tupled((file_path,
+                        actualLastUpdateYear,
+                        gitLastUpdateYear,
+                        actualCreationYear,
+                        gitCreationYear))
+      }
+    }
+    data
+  }
+
+  def fixWrongDates(): Boolean = {
+    var check = false
+    val files = analyzeDates()
+    files.foreach { file =>
+      val actual_file = readFile(file.pathToFile)
+      var new_file = ""
+      if (file.actualLastUpdateYear.forall(_.isDigit)) {
+        if (file.ifUpdateDateWrong()) {
+          new_file = actual_file.
+                       replace(
+                        s"${file.actualCreationYear}-${file.actualLastUpdateYear}",
+                        s"${file.actualCreationYear}-${file.gitLastUpdateYear}"
+                       )
+        }
+      }
+      if(new_file != "") {
+        scala.tools.nsc.io.File(file.pathToFile)("UTF-8").writeAll(new_file)
+        check = true
+      }
+    }
+    check
+  }
+
+  def resultAnalyzeDates(): Boolean = {
+    val data = analyzeDates
+    var check = false
+    data.foreach {file =>
+      if (file.ifUpdateDateWrongStr()) {
+        if(file.actualLastUpdateYear.forall(_.isDigit)) {
+          println(file + " wrong date ")
+          check = true
+        }
+      }
+    }
+    if (check) println("Oops...We have some error dates")
+    else println ("Everything is ok")
+    check
+  }
+
+  def main(args: Array[String]) {
```

Comment:
If we have found wrong date = true returns
all is ok and we don't find nothing = false returns.